### PR TITLE
converted InteropEventEmitter java file to kotlin

### PR DESCRIPTION
**Pull request by stanwolverine**
## Summary:

Helping in [Kotlin-ify React Native tests](https://github.com/facebook/react-native/issues/37708), Converted [react/fabric/interop/InteropEventEmitterTest.java](https://github.com/facebook/react-native/tree/main/packages/react-native/ReactAndroid/src/test/java/com/facebook/react/fabric/interop/InteropEventEmitterTest.java) to kotlin file

## Changelog:

[INTERNAL] [CHANGED] - `InteropEventEmitterTest.java` file migrated from java to kotlin

## Test Plan:

Tests should pass
```
./gradlew :packages:react-native:ReactAndroid:test
```

### CODE REVIEW

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/fabric/interop/InteropEventEmitterTest.kt [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import org.junit.Assert.assertNotNull
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.RuntimeEnvironment
+
+@RunWith(RobolectricTestRunner::class)
+class InteropEventEmitterTest {
+  lateinit var mReactContext: ReactContext
```
Comment (mdvacca) [inline packages/react-native/ReactAndroid/src/test/java/com/facebook/react/fabric/interop/InteropEventEmitterTest.kt:23]: can you remove m prefix?

```
lateinit var reactContext: ReactContext
```

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/fabric/interop/InteropEventEmitterTest.kt [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  lateinit var mEventDispatcher: FakeEventDispatcher
```
Comment (mdvacca) [inline packages/react-native/ReactAndroid/src/test/java/com/facebook/react/fabric/interop/InteropEventEmitterTest.kt:24]: same comment as above

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/fabric/interop/InteropEventEmitterTest.kt (hunk 1 above)
Comment (stanwolverine) [reply packages/react-native/ReactAndroid/src/test/java/com/facebook/react/fabric/interop/InteropEventEmitterTest.kt]: Done ✅

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/fabric/interop/InteropEventEmitterTest.kt (hunk 2 above)
Comment (stanwolverine) [reply packages/react-native/ReactAndroid/src/test/java/com/facebook/react/fabric/interop/InteropEventEmitterTest.kt]: Done ✅

**Review by mdvacca — APPROVED** [review]
lgtm

**Pull request closed without merging**